route-rs-runtime/src/link/primitive/fork_link.rs

```diff
@@ -149,22 +151,21 @@ impl<P> Drop for ForkIngressor<P> {
 }
 
 impl<P: Send + Clone> Future for ForkIngressor<P> {
-    type Item = ();
-    type Error = ();
+    type Output = ();
 
     /// If any of the channels are full, we await that channel to clear before processing a new packet.
-    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
```

Comment:
Why declare `PacketStream`s `Unpin` if packets are `Pin`?

Reply:
I'm not asking to be snarky, I genuinely don't know.

Reply:
Packets have no Pin requirement. In your example, ForkIngressor the object must be Pinned before one may call poll on it. 
For example
`
let cx = Context; //We got this variable from somewhere, not important for this example.  
let forkingressor = ForkIngressor::new() //lets pretend this works
forkingressor.poll(cx); //Compiler error: method not found
Pin::new(&mut forkingressor).poll(cx);  //Happy rustc
`